### PR TITLE
[DEN-72] GO SDK: Delete Custom Rule Set #23

### LINKED ISSUE #23: [DEN-72] GO SDK: Delete Custom Rule Set

### PR DESCRIPTION
**Pull request by ssaluja4**
Closes #23

### CODE REVIEW

File: edgecast/waf/common_types.go [hunk 1]
```diff
@@ -19,6 +19,19 @@ type AddRuleResponse struct {
 	Errors []WAFError
 }
 
+// DeleteRuleResponse contains the response from the WAF API when deleting a new rule
```
Comment (changgyu-oh) [inline edgecast/waf/common_types.go:22]: Maybe we should use the struct composite we learn yesterday.

File: edgecast/waf/custom_rule.go [hunk 2]
```diff
@@ -247,3 +256,24 @@ func (svc *WAFService) GetAllCustomRuleSets(accountNumber string) ([]CustomRuleS
 
 	return *customRuleSets, nil
 }
+
+// Deletes a custom rule.
+func (svc *WAFService) DeleteCustomRuleSet(accountNumber string, customRuleId string) (*DeleteCustomRuleResponse, error) {
```
Comment (changgyu-oh) [inline edgecast/waf/custom_rule.go:261]: We can discuss about naming convention but golang recommend to use customRuleID instead of customRuleId.

**Review by changgyu-oh — COMMENTED** [review]
I made couple comments.

File: edgecast/waf/common_types.go (hunk 1 above)
Comment (stevenpaz) [reply edgecast/waf/common_types.go:22]: Agreed with @changgyu-oh , we could have a base `WAFResponse` struct with `Succcess` `Status` and `Errors`, then `AddRuleResponse` can add `ID` and `WAFResponse` as properties

File: edgecast/waf/custom_rule.go [hunk 3]
```diff
@@ -10,6 +10,15 @@ import (
 	This file contains operations and types specific to WAF Custom Rule Sets.
 */
 
+// DeleteCustomRuleResponse contains the response from the WAF API when deleting a new rule
+type DeleteCustomRuleResponse struct {
```
Comment (stevenpaz) [inline edgecast/waf/custom_rule.go:14]: This may no longer be needed if the base struct is implemented as discussed above

**Review by stevenpaz — CHANGES_REQUESTED** [review]

File: edgecast/waf/common_types.go (hunk 1 above)
Comment (ssaluja4) [reply edgecast/waf/common_types.go]: Created WAFResponse Class and updated response classes to include the base class

File: edgecast/waf/custom_rule.go (hunk 2 above)
Comment (ssaluja4) [reply edgecast/waf/custom_rule.go]: agreed. updated to golang conventions.

File: edgecast/waf/custom_rule.go (hunk 3 above)
Comment (ssaluja4) [reply edgecast/waf/custom_rule.go:14]: Updated to include WAF response struct + one additional property needed

**Review by stevenpaz — APPROVED** [review]

**Pull request merged**